### PR TITLE
Option to use usesDrawViewHierarchyInRect:afterScreenUpdates:

### PR DESCRIPTION
**Pull request by nscoding**
There have been several cases (#70 #91 #92) where the view needs to be added in a visible `UIWindow` to display things correctly. Calling `fb_imageForLayer:` with `view.layer` is not enough in these cases.

- The new property (`usesDrawViewHierarchyInRect`) will use `drawViewHierarchyInRect:afterScreenUpdates:` like Apple suggests [here](https://developer.apple.com/library/ios/qa/qa1817/_index.html). 
  - The view is added in a `UIWindow` if needed.
- Snapshot tests are also improved on the demo application.

### CODE REVIEW

File: FBSnapshotTestCaseDemo/FBSnapshotTestCaseDemoTests/FBSnapshotTestCaseDemoTests.m [hunk 1]
```diff
@@ -26,10 +25,39 @@ - (void)setUp
...
+  if ([UIVisualEffect class]) {
+    UIVisualEffect *blurEffect = [UIBlurEffect effectWithStyle:UIBlurEffectStyleLight];
+    UIVisualEffectView *visualEffectView = [[UIVisualEffectView alloc] initWithEffect:blurEffect];
+    UIView *bluredRedView = [[UIView alloc] initWithFrame:CGRectMake(0, 0, 40, 40)];
+    bluredRedView.backgroundColor = [UIColor redColor];
+    visualEffectView.frame = bluredRedView.bounds;
+    [bluredRedView addSubview:visualEffectView];
+    
+    self.usesDrawViewHierarchyInRect = YES;
+    FBSnapshotVerifyView(bluredRedView, nil);
```
Comment (Grubas7) [inline FBSnapshotTestCaseDemo/FBSnapshotTestCaseDemoTests/FBSnapshotTestCaseDemoTests.m]: You won't see on snapshot blur/vibrant effect if both view will have same frames.
I would recommend to change line 44 to:
visualEffectView.frame = CGRectMake(0, 0, 20, 40);

File: FBSnapshotTestCaseDemo/FBSnapshotTestCaseDemoTests/FBSnapshotTestCaseDemoTests.m [hunk 2]
```diff
@@ -26,10 +25,39 @@ - (void)setUp
...
+}
+
+- (void)testViewSnapshotWithVisualEffects
+{
+  if ([UIVisualEffect class]) {
+    UIVisualEffect *blurEffect = [UIBlurEffect effectWithStyle:UIBlurEffectStyleLight];
+    UIVisualEffectView *visualEffectView = [[UIVisualEffectView alloc] initWithEffect:blurEffect];
+    UIView *bluredRedView = [[UIView alloc] initWithFrame:CGRectMake(0, 0, 40, 40)];
+    bluredRedView.backgroundColor = [UIColor redColor];
+    visualEffectView.frame = bluredRedView.bounds;
```
Comment (Grubas7) [inline FBSnapshotTestCaseDemo/FBSnapshotTestCaseDemoTests/FBSnapshotTestCaseDemoTests.m]: You won't see on snapshot blur/vibrant effect if both view will have same frames.
I would recommend to change to:
visualEffectView.frame = CGRectMake(0, 0, 20, 40);

File: FBSnapshotTestCaseDemo/FBSnapshotTestCaseDemoTests/FBSnapshotTestCaseDemoTests.m (hunk 1 above)
Comment (Grubas7) [inline FBSnapshotTestCaseDemo/FBSnapshotTestCaseDemoTests/FBSnapshotTestCaseDemoTests.m]: You won't see on snapshot blur/vibrant effect if both view will have same frames.
I would recommend to change to:
UIView *redView = [[UIView alloc] initWithFrame:CGRectMake(0, 0, 20, 40)];
redView.backgroundColor = [UIColor redColor];
visualEffectView.frame = CGRectMake(0, 0, 40, 20);

UIView *parentView = [[UIView alloc] initWithFrame:CGRectMake(0, 0, 40, 40)];
[parentView addSubview:redView];
[parentView addSubview:visualEffectView];

Comment (nscoding) [inline FBSnapshotTestCaseDemo/FBSnapshotTestCaseDemoTests/FBSnapshotTestCaseDemoTests.m]: :+1:

**Pull request merged**